pkg/util/push/push.go

```diff
@@ -42,6 +45,12 @@ func Handler(maxRecvMsgSize int, sourceIPs *middleware.SourceIPExtractor, push F
 		}
 
 		if _, err := push(ctx, &req.WriteRequest); err != nil {
+			if errors.Is(err, context.Canceled) {
```

Comment:
>We should continue to return 500 if the remote-timeout is hit

I didn't get the issue that needs to be ensured here, because there is no change(the PR) in the error reporting behavior of when the request timeout.

Reply:
@wgliang a test for DeadlineExceeded would be nice to have.

Reply:
> @wgliang a test for DeadlineExceeded would be nice to have.

done

Reply:
I didn't look into this in detail, but still some middle-level contexts could be canceled(not deadline exceeded), so we may need to ensure that.

Reply:
> I didn't look into this in detail, but still some middle-level contexts could be canceled(not deadline exceeded), so we may need to ensure that.

If there is none, any changes after this may add such behavior(cancel the context), which brings complexity and mental burden. We may need a solution for it.

Reply:
> > I didn't look into this in detail, but still some middle-level contexts could be canceled(not deadline exceeded), so we may need to ensure that.
> 
> If there is none, any changes after this may add such behavior(cancel the context), which brings complexity and mental burden. We may need a solution for it.

Emm, can you construct a test here to hit the issue you mentioned?